File: demos/tx/java-demo/theta-tx-demo/src/main/java/org/theta/tx/types/TxOutput.java
package org.theta.tx.types;

import java.math.BigInteger;
import org.ethereum.util.RLP;

public final class TxOutput {

    public byte[] address;
    public Coins  coins;

    public TxOutput(byte[] address, BigInteger thetaWei, BigInteger tfuelWei) {
        this.address = address;
        this.coins = new Coins(thetaWei, tfuelWei);
    }

    public byte[] rlpEncode() {
        byte[] rlpEncoded = RLP.encodeList(
            RLP.encode(this.address),
            this.coins.rlpEncode());
        return rlpEncoded;
    }
}

File: demos/tx/javascript-demo/theta-tx-demo/src/tx/types/txoutput.js
import {Coins} from './coins';

export class TxOutput {
    constructor(address, thetaWei, tfuelWei) {
        this.address = address;
        this.coins = new Coins(thetaWei, tfuelWei);
    }

    rlpInput(){
        let rplInput = [
            this.address.toLowerCase(),
            this.coins.rlpInput()
        ];

        return rplInput;
    }
}